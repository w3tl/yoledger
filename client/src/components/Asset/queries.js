import gql from 'graphql-tag';
import fragments from './fragments';

export const LIST_QUERY = gql`
query AssetsQuery {
  assets: accounts(type: ASSET) {
    ...AssetFormAccount
  }
}
${fragments.asset}
`;

export const ADD_MUTATION = gql`
mutation addAccount($input: AddAccountInput!) {
  addAccount(input: $input) {
    account {
      ...AssetFormAccount
    }
  }
}
${fragments.asset}
`;
