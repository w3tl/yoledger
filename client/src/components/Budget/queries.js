import gql from 'graphql-tag';

export const BODY_QUERY = gql`
query BudgetPeriodsQuery($dateStart: Date!, $dateEnd: Date!) {
  budgets(dateStart: $dateStart, dateEnd: $dateEnd) {
    accounts {
      name
    }
  }
}
`;

export const ROW_QUERY = gql`
query BudgetQuery($account: String!, $dateStart: Date!, $dateEnd: Date!) {
  budget(account: $account, dateStart: $dateStart, dateEnd: $dateEnd) {
    id
    date
    amount
  }
}
`;

export const UPDATE_MUTATION = gql`
mutation upsertBudget($input: UpsertBudgetInput!) {
  upsertBudget(input: $input) {
    success
    budget {
      id
      account {
        id
        name
      }
      date
      amount
    }
  }
}
`;
