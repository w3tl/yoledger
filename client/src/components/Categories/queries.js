import gql from 'graphql-tag';
import fragments from './fragments';

export const LIST_QUERY = gql`
query CategoriesQuery {
  expenses: accounts(type: EXPENSE) {
    ...Category
  }
  incomes: accounts(type: INCOME) {
    ...Category
  }
}
${fragments.category}
`;

export const ADD_MUTATION = gql`
mutation addAccount($input: AddAccountInput!) {
  addAccount(input: $input) {
    account {
      ...Category
    }
  }
}
${fragments.category}
`;
