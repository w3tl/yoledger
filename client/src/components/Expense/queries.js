import gql from 'graphql-tag';
import fragments from './fragments';

export const LIST_QUERY = gql`
query ExpensesQuery {
  accounts(type: EXPENSE) {
    ...ExpenseAccount
  }
}
${fragments.expense}
`;

export const FORM_QUERY = gql`
query ExpenseQuery($id: ID!) {
  account(id: $id) {
    ...ExpenseAccount
  }
}
${fragments.expense}
`;

export const ADD_MUTATION = gql`
mutation addAccount($input: AddAccountInput!) {
  addAccount(input: $input) {
    account {
      ...ExpenseAccount
    }
  }
}
${fragments.expense}
`;
