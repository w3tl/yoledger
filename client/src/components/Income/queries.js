import gql from 'graphql-tag';
import fragments from './fragments';

export const LIST_QUERY = gql`
query IncomesQuery {
  accounts(type: INCOME) {
    ...IncomeAccount
  }
}
${fragments.income}
`;

export const FORM_QUERY = gql`
query IncomeQuery($id: ID!) {
  account(id: $id) {
    ...IncomeAccount
  }
}
${fragments.income}
`;

export const ADD_MUTATION = gql`
mutation addAccount($input: AddAccountInput!) {
  addAccount(input: $input) {
    account {
      ...IncomeAccount
    }
  }
}
${fragments.income}
`;
