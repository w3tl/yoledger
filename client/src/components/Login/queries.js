import gql from 'graphql-tag';

export const SIGNIN_MUTATION = gql`
mutation signin($login: String!, $password: String!) {
  signin(login: $login, password: $password) {
    token
  }
}
`;

export const SIGNOUT_MUTATION = gql`
mutation signout($token: String) {
  signout(token: $token) {
    success
  }
}
`;
