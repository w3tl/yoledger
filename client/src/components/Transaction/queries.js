import gql from 'graphql-tag';
import fragments from './fragments';

export const FORM_QUERY = gql`
query TransactionQuery($id: ID!) {
  transaction(id: $id) {
    ...TransactionFormTransaction
  }
}
${fragments.transaction}
`;

export const LIST_QUERY = gql`
query TransactionsQuery($dateStart: Date!, $dateEnd: Date, $page: Int, $itemsPerPage: Int) {
  transactions(dateStart: $dateStart, dateEnd: $dateEnd, page: $page, itemsPerPage: $itemsPerPage) {
    ...TransactionFormTransaction
  }
}
${fragments.transaction}
`;

export const ADD_MUTATION = gql`
mutation addTransaction($input: AddTransactionInput!) {
  addTransaction(input: $input) {
    transaction {
      ...TransactionFormTransaction
    }
  }
}
${fragments.transaction}
`;

export const DELETE_MUTATION = gql`
mutation deleteTransaction($id: ID!) {
  deleteTransaction(id: $id) {
    success
    id
  }
}
`;

export const UPDATE_MUTATION = gql`
mutation updateTransaction($id: ID!, $input: UpdateTransactionInput!) {
  updateTransaction(id: $id, input: $input) {
    transaction {
      ...TransactionFormTransaction
    }
  }
}
${fragments.transaction}
`;

export const GET_PAGE_SETTINGS = gql`
query {
  settings @client {
    transaction {
      period
      date
    }
  }
}
`;

export const CHANGE_PERIOD = gql`
  mutation changePeriod($period: Period, $date: String) {
    changeTransactionPagePeriod(period: $period, date: $date) @client {
      period
      date
    }
  }
`;
