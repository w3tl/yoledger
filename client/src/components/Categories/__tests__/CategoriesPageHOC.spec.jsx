/* eslint-disable react/prop-types */
import React from 'react';
import { mount } from 'enzyme';
import { LIST_QUERY } from '../queries';
import RawCategoriesPage, { updateAfterAdd } from '../CategoriesPageHOC';
import {
  wait, withProvider, testLoadingState, client,
} from '../../testHelpers';


const CategoriesPage = withProvider((props) => {
  props.client.writeQuery({
    query: LIST_QUERY,
    data: { expenses: [], incomes: [] },
  });
  return <RawCategoriesPage {...props} />;
});

describe('CategoriesPageHOC', () => {
  testLoadingState(<CategoriesPage />);

  describe('updateAfterAdd', () => {
    it('should update expense cache', () => {
      const writeQuery = jest.fn();
      const cache = {
        readQuery: jest.fn().mockReturnValue({ expenses: [], incomes: [] }),
        writeQuery,
      };
      const data = { addAccount: { account: { type: 'EXPENSE' } } };
      updateAfterAdd(cache, { data });
      expect(writeQuery.mock.calls[0][0].data).toMatchSnapshot('writeQuery');
    });

    it('should update income cache', () => {
      const writeQuery = jest.fn();
      const cache = {
        readQuery: jest.fn().mockReturnValue({ expenses: [], incomes: [] }),
        writeQuery,
      };
      const data = { addAccount: { account: { type: 'INCOME' } } };
      updateAfterAdd(cache, { data });
      expect(writeQuery.mock.calls[0][0].data).toMatchSnapshot('writeQuery');
    });
  });

  it('should contain props', async () => {
    const wrapper = mount(<CategoriesPage />);
    await wait();
    wrapper.update();
    const form = wrapper.find('CategoriesPage');
    expect(form.prop('expenses')).toBeDefined();
    expect(form.prop('incomes')).toBeDefined();
  });

  it('should successfully add expense', async () => {
    const wrapper = mount(<CategoriesPage defaultActiveIndex={2} />);
    await wait();
    wrapper.update();
    wrapper.find('input[name="name"]').first().simulate('change', { target: { value: 'new expense' } });
    wrapper.find('form').simulate('submit');
    await wait(1);
    wrapper.update();
    const { expenses } = client.readQuery({
      query: LIST_QUERY,
    });
    expect(expenses).toHaveLength(1);
  });
});
