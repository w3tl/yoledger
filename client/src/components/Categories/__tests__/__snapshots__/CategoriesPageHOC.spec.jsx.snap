// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesPageHOC updateAfterAdd should update expense cache: writeQuery 1`] = `
Object {
  "expenses": Array [
    Object {
      "type": "EXPENSE",
    },
  ],
  "incomes": Array [],
}
`;

exports[`CategoriesPageHOC updateAfterAdd should update income cache: writeQuery 1`] = `
Object {
  "expenses": Array [],
  "incomes": Array [
    Object {
      "type": "INCOME",
    },
  ],
}
`;
